tests/src/core/testqgsmeshlayer.cpp

```diff
@@ -871,6 +887,7 @@ void TestQgsMeshLayer::test_reload_extra_dataset()
   //test dataset presence
   QCOMPARE( layer.dataProvider()->extraDatasets().count(), 2 );
   QCOMPARE( layer.dataProvider()->datasetGroupCount(), 3 );
+  QCOMPARE( layer.datasetGroupTreeRootItem()->childCount(), 3 ); //dataset group tree item still have all dataset groups
 
   //copy the qad_and_triangle_vertex_vector.dat to the temporary testFile
```

Comment:
Could you please also add tests for the newly added dataset group tree item class?
- verify that getters/setters work correctly with a test tree
- making sure that load/save works (the tree will be the same as before, including group indexes, enabled flags, names)